server/spectatord.cc

```diff
@@ -246,23 +246,37 @@ auto get_measurement(std::string_view measurement_str, std::string* err_msg)
       pos = v_pos;
     }
   }
+
   ++pos;
   auto value_str = measurement_str.begin() + pos;
   char* last_char = nullptr;
-  auto value = std::strtod(value_str, &last_char);
+  double value_d;
+  uint64_t value_u;
+  if (type == 'C' || type == 'X') {
+    value_u = std::strtoull(value_str, &last_char, 10);
```

Comment:
Note, on the java side we support monotonic [longs](https://www.javadoc.io/static/com.netflix.spectator/spectator-api/1.7.13/com/netflix/spectator/api/patterns/PolledMeter.Builder.html#monitorMonotonicCounter-T-java.util.function.ToLongFunction-) and [doubles](https://www.javadoc.io/static/com.netflix.spectator/spectator-api/1.7.13/com/netflix/spectator/api/patterns/PolledMeter.Builder.html#monitorMonotonicCounterDouble-T-java.util.function.ToDoubleFunction-). The main use-case for double was to support arbitrary unit conversions so users could keep a base unit. For example instead of needing a special type like `X` they could divide by 1000.0 and use the normal monotonic counter. Bytes is another example where there can be numerous factors (bits, powers of 1000, powers of 1024, etc).